shotover-proxy/src/transforms/tee.rs

```diff
@@ -26,29 +26,65 @@ pub struct Tee {
 pub enum ConsistencyBehavior {
     Ignore,
     FailOnMismatch,
-    SubchainOnMismatch(Vec<TransformsConfig>),
```

Comment:
Why didnt this work?
This approach seems simpler and better UX.

Reply:
Because we can't use `behavior: SubchainOnMismatch: failchain:` in YAML

Reply:
That sounds wrong to me, what happens if you serialize ConsistencyBehaviour::SubchainOnMismatch to yaml